core/src/main/java/io/grpc/internal/PickFirstLoadBalancerProvider.java

```diff
@@ -54,6 +54,8 @@ public LoadBalancer newLoadBalancer(LoadBalancer.Helper helper) {
   @Override
   public ConfigOrError parseLoadBalancingPolicyConfig(
       Map<String, ?> rawLoadBalancingPolicyConfig) {
-    return ConfigOrError.fromConfig(NO_CONFIG);
```

Comment:
Shouldn't the new versus old behavior be conditional on enablePickFirstConfig?

Reply:
This made me think of the whole approach. I added the flag checking here in the provider and added a test for it as well. In the actual LB I just apply shuffling if the configuration comes down and leave the provider to gate it with the flag. I think that is more straightforward.